core/domain/fs_domain.py

```diff
@@ -325,7 +325,8 @@ def commit(self, filepath, raw_bytes, mimetype=None):
             raw_bytes: str. The content to be stored in the file.
             mimetype: str. The content-type of the file.
         """
-        raw_bytes = python_utils.convert_to_bytes(raw_bytes)
+        if mimetype != 'application/octet-stream':
```

Comment:
Its not the issue with bytes but with `encode()` function. The `convert_to_bytes` method encodes as utf-8 if input is of `str` type and it fails at that step since bytes are still of `str` type in python2 (this wouldn't be a problem with python 3 though).

Reply:
Ah, I see. Can you find a way to do this such that the py2/py3 variation is encapsulated fully within python_utils.py, so that, when we switch from py2 to py3, no change is needed in this file?

Reply:
The current logic won't require any changes if we move from py2 to py3. This one will work because protobuf `SerializeToString()` method directly yields bytes data (which takes care of py2 vs py3) and hence we can just go ahead and store it without requiring `convert_to_bytes`. The `convert_to_bytes` is required to store textual data but when file is opened in `*b (binary)` mode. In that case, we can't store text directly and need to bytes conversion but here since we already have bytes we don't need to do it again. So this code works on both py2 and py3.